File: scout_discord_day3.py
#!/usr/bin/env python3
"""Scout Indie Park #vos-créations for Day 3 — print lead signals only."""
from __future__ import annotations

import json
import urllib.request
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
cfg = json.loads((ROOT / "data" / "config.json").read_text(encoding="utf-8"))
token = cfg["discord_token"]
cid = "1064526595030782043"
skip = {
    "zebra3d",
    "Mephistase",
    "color500",
    "Olonnais",
    "Razjinh",
    "papapoule75",
    "valera7623",
}
keys = (
    "steam",
    "local",
    "traduc",
    "transl",
    "csv",
    "wishlist",
    "feedback",
    "demo",
    "alpha",
    "playtest",
    "showcase",
    "release",
    "avis",
    "retour",
    "multilang",
    "langue",
    "itch",
    "test",
)

req = urllib.request.Request(
    f"https://discord.com/api/v10/channels/{cid}/messages?limit=50",
    headers={"Authorization": f"Bot {token}", "User-Agent": "LocForgeScout/1.0"},
)
try:
    with urllib.request.urlopen(req, timeout=25) as r:
        msgs = json.loads(r.read().decode())
except Exception as e:
    print("ERR", type(e).__name__, e)
    raise SystemExit(1)

print("OK", len(msgs))
for m in msgs:
    a = m.get("author") or {}
    un = a.get("username") or ""
    if un in skip or a.get("bot"):
        continue
    content = (m.get("content") or "").replace("\n", " ")
    low = content.lower()
    signals = any(k in low for k in keys) or "store.steampowered" in low
    if signals or len(content) > 80:
        mid = m["id"]
        url = f"https://discord.com/channels/1021378341267320872/{cid}/{mid}"
        print("---")
        print("id", mid)
        print("user", un)
        print("sig", int(signals))
        print("url", url)
        print("text", content[:280])

File: scout_discord_day3.js
const fs = require("fs");
const path = require("path");
const https = require("https");

const root = __dirname;
const cfg = JSON.parse(fs.readFileSync(path.join(root, "data", "config.json"), "utf8"));
const token = cfg.discord_token;
const cid = "1064526595030782043";
const skip = new Set([
  "zebra3d",
  "Mephistase",
  "color500",
  "Olonnais",
  "Razjinh",
  "papapoule75",
  "valera7623",
]);
const keys = [
  "steam",
  "local",
  "traduc",
  "transl",
  "csv",
  "wishlist",
  "feedback",
  "demo",
  "alpha",
  "playtest",
  "showcase",
  "release",
  "avis",
  "retour",
  "multilang",
  "langue",
  "itch",
  "test",
];

const req = https.request(
  {
    hostname: "discord.com",
    path: `/api/v10/channels/${cid}/messages?limit=50`,
    method: "GET",
    headers: {
      Authorization: `Bot ${token}`,
      "User-Agent": "LocForgeScout/1.0",
    },
  },
  (res) => {
    let body = "";
    res.on("data", (c) => (body += c));
    res.on("end", () => {
      if (res.statusCode !== 200) {
        console.log("ERR", res.statusCode, body.slice(0, 300));
        process.exit(1);
      }
      const msgs = JSON.parse(body);
      console.log("OK", msgs.length);
      for (const m of msgs) {
        const a = m.author || {};
        const un = a.username || "";
        if (skip.has(un) || a.bot) continue;
        const content = (m.content || "").replace(/\n/g, " ");
        const low = content.toLowerCase();
        const signals =
          keys.some((k) => low.includes(k)) || low.includes("store.steampowered");
        if (signals || content.length > 80) {
          const url = `https://discord.com/channels/1021378341267320872/${cid}/${m.id}`;
          console.log("---");
          console.log("id", m.id);
          console.log("user", un);
          console.log("sig", signals ? 1 : 0);
          console.log("url", url);
          console.log("text", content.slice(0, 280));
        }
      }
    });
  }
);
req.on("error", (e) => {
  console.log("ERR", e.message);
  process.exit(1);
});
req.end();
